fvm/state/state.go

```diff
@@ -197,12 +198,18 @@ func (s *State) MergeState(other *State) error {
 	return s.checkMaxInteraction()
 }
 
-// UpdatedAddresses returns a list of addresses that were updated (at least 1 register update)
+// UpdatedAddresses returns a sorted list of addresses that were updated (at least 1 register update)
 func (s *State) UpdatedAddresses() []flow.Address {
 	addresses := make([]flow.Address, 0, len(s.updatedAddresses))
+
+	// sort list wile constructing it
 	for k := range s.updatedAddresses {
-		addresses = append(addresses, k)
+		i := sort.Search(len(addresses), func(i int) bool { return addresses[i].Hex() >= k.Hex() })
```

Comment:
I'm afraid this will be used a lot (for every transaction?) and using `.Hex()` isn't best performance-wide. `bytes.Compare` instead?

Reply:
I was going to suggest similar thing, using bytes.Compare over slices or conversion to uint64 and compare considering the number of bytes of addresses

Reply:
I would rather do byte comparison... converting to uint introduces an element that could fail.

Reply:
> converting to uint introduces an element that could fail.

Not really - one of the Go quirks I learned recently, it will silently truncate the value https://golang.org/ref/spec#Conversions